person.rb

```diff
@@ -1,26 +1,33 @@
 require './corrector'
+require './classroom'
 
 class Person
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rentals = []
   end
 
-  attr_accessor :name, :age
-  attr_reader :id
-
   def of_age?
     @age >= 18
   end
 
   def can_use_services?
-    parent_permission || of_age
+    @parent_permission || of_age
   end
 
   def validate_name
     @name = @corrector.correct_name(@name)
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.classroom = self
+  end
```

Comment:
- [ ] This `add_rental` method will also need to be refactored as above. The only difference here is that it will need **`book`** and **`date`** as parameters.

Don't forget to add **`self`** which refers to the `Person` renting the book when creating the Rental.